### DISCUSSION TITLE
What is the best way to target html attribute?

### DISCUSSION THREAD

**Discussion by michael-land** [tailwindlabs/tailwindcss#6041] [Help] [answered]
Hi, what is the best practice when targeting element with data attribute?

for example, when a Tab is selected, it adds a `data-state="selected"` attribute `<div data-state="open">`, how do i style such element without write custom css?

**Comment by olets** [comment, marked as answer]
You can write a "basic variant" plugin https://tailwindcss.com/docs/plugins#adding-variants. The result will be to support things like

```html
<div class="state-open:text-red-400" data-state="selected">this is not red</div>
<div class="state-open:text-red-400" data-state="open">this is red</div>
```
